app/interactors/sync.rb

```diff
@@ -90,4 +84,61 @@ def skip_updating
     self.class.skip_updating
   end
 
+  # Creates or updates a single model in the database and sets it's id.
+  #
+  # @param model [Sequel::Model] instance to sync
+  def upsert_model(model)
+    model_hash = model.to_hash
+    columns_to_update = model_hash.keys.reject do |k|
+      matching_attributes.include?(k) || skip_updating.include?(k)
+    end
+    upsert_options = { target: matching_attributes }
+    unless columns_to_update.empty?
+      update_clause = columns_to_update.map { |key| [ key.to_sym, "excluded__#{key}".to_sym ] }.to_h
+      timestamps = update_timestamps(columns_to_update)
+      upsert_options[:update] = update_clause.merge(timestamps) { |key, oldval, newval| oldval }
+    end
+    model_insert_clause = model_hash.merge(insert_timestamps) { |key, oldval, newval| oldval }
+    model.id = model_class.dataset.insert_conflict(upsert_options).insert(model_insert_clause)
+    model.instance_variable_set(:@new, false)
+  end
+
+  # Generates created_at and updated_at fields for insert clause if model has
+  # either created_at or updated_at column.
+  #
+  # @return [Hash] with timestamps
+  def insert_timestamps
+    timestamps = {}
+    if model_class.columns.include?(:created_at)
+      timestamps = timestamps.merge({ created_at: Sequel.function(:NOW) })
```

Comment:
Why so complicated?

`timestamps[:created_at] = Sequel.function(:NOW)` should do the same thing, shouldn’t it?